Using synth to build armv6 binaries from amd64 host

I'm attempting to use synth to build armv6 ports.

I have:
- built a FreeBSD armv6 tree with make buildworld installworld
- used binmiscctl to activate qemu-arm-static
- successfully built a simple port (ports-mgmt/pkg) within the chroot'd armv6 tree.

I have configured a profile for synth, with the following snippet from synth.ini
[armv6]
Operating_system= FreeBSD
Directory_packages= /var/synth/armv6
Directory_repository= /var/synth/armv6/All
Directory_portsdir= /usr/ports
Directory_options= /var/db/ports
Directory_distfiles= /var/distfiles
Directory_buildbase= /usr/obj/synth-live
Directory_logs= /var/log/synth-armv6
Directory_ccache= disabled
Directory_system= /home/jonc/work/freebsd-arm/arm-root
Number_of_builders= 4
Max_jobs_per_builder= 3
Tmpfs_workdir= true
Tmpfs_localbase= true
Display_with_ncurses= true
leverage_prebuilt= false

However, running "env SYNTHPROFILE=armv6 synth build ports-mgmt/pkg" results in a failure during due to a BADARCH reference:
...
===>  Configuring for pkg-1.10.0_2
configure: loading site script /xports/Templates/config.site
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... (cached) /bin/mkdir -p
checking for gawk... (cached) /usr/bin/awk   
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking for style of include used by make... GNU
checking for gcc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking whether cc understands -c and -o together... yes
checking dependency style of cc... gcc3
checking how to run the C preprocessor... cpp
checking for grep that handles long lines and -e... (cached) /usr/bin/grep
checking for egrep... (cached) /usr/bin/egrep
checking for ANSI C header files... (cached) yes
checking for sys/types.h... (cached) yes
checking for sys/stat.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for memory.h... (cached) yes
checking for strings.h... (cached) yes
checking for inttypes.h... (cached) yes
checking for stdint.h... (cached) yes
checking for unistd.h... (cached) yes
checking for minix/config.h... (cached) no
checking whether it is safe to define __EXTENSIONS__... yes
checking for cc option to accept ISO C99... none needed
checking build system type... Invalid configuration `BADARCH-portbld-freebsd11.0': machine `BADARCH-portbld' not recognized
configure: error: /bin/sh ./config.sub BADARCH-portbld-freebsd11.0 failed
===>  Script "configure" failed unexpectedly.
...

I have noticed in the log that there is a `ARCH=BADARCH`, just after including armv6-make.conf, but I do not seem to be able to override it.

How does synth determine the build-architecture? From within the chroot'd armv6 tree, I get the correct values:
root@helios:/ # uname -m -p
arm armv6
root@helios:/ # sysctl hw.machine_arch
hw.machine_arch: armv6

Is it possible for me to fix the BADARCH reference?

--- a/src/replicant-platform.adb
+++ b/src/replicant-platform.adb
@@ -187,6 +187,8 @@ package body Replicant.Platform is
            arch (arch'First .. arch'First + 4) = "AMD64"
          then
             return "x86:64";
+         elsif arch (arch'First .. arch'First + 2) = "ARM" then
+            return "armv6:32:el:eabi:softfp";
          elsif arch = "Intel 80386" then
             return "x86:32";
          else
@@ -200,6 +202,8 @@ package body Replicant.Platform is
            arch (arch'First .. arch'First + 4) = "AMD64"
          then
             return "amd64";
+         elsif arch (arch'First .. arch'First + 2) = "ARM" then
+            return "armv6";
          elsif arch = "Intel 80386" then
             return "i386";
          else

--- a/src/replicant-platform.adb
+++ b/src/replicant-platform.adb
@@ -148,6 +148,8 @@ package body Replicant.Platform is
                when solaris   => return "x86_64";
                when unknown   => return badarch;
             end case;
+         elsif arch (arch'First .. arch'First + 2) = "ARM" then
+            return "armv6";
          elsif arch = "Intel 80386" then
             return "i386";
          else

